packages/react-storage/src/components/StorageBrowser/views/LocationActionView/useHandleUpload.ts

```diff
@@ -135,10 +138,20 @@ export function useHandleUpload({
         };
       });
 
+      tracker.recordTasksToProcess(newTasks);
       return mergeSelectedTasks(prevTasks, newTasks);
     });
   };
 
+  const handleCancelAll = React.useCallback(() => {
+    tasks.forEach(({ cancel }) => (isFunction(cancel) ? cancel() : undefined));
+  }, [tasks]);
+
+  const uploadFatalError = React.useCallback(() => {
+    handleCancelAll();
+    throw new Error('Upload validation failed');
```

Comment:
How is this error prevented from causing an uncaught exception?

Reply:
This error should only happen when the requested task count does not match completed task count. It would never happen under any normal circumstances.